From de1722f4b4c3d5e48c5253201baba6e7bbb20579 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 23 Jun 2022 11:10:31 +0200
Subject: [PATCH 036/120] arm: stm32mp: simply reset the TAMP_COPRO_STATE
 backup register

The tamp register should be reset instead of being
set to a specific value.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 3f31fca325fd05b50940c9f19ca6bfd7a41fe648 arm: stm32mp: simply reset the TAMP_COPRO_STATE backup register]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I19c80fea5f8d9e7ef1ee32ec5a49bb520e448053
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/315246
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Michel JAOUEN <michel.jaouen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 arch/arm/mach-stm32mp/stm32mp1/stm32mp15x.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-stm32mp/stm32mp1/stm32mp15x.c b/arch/arm/mach-stm32mp/stm32mp1/stm32mp15x.c
index 4f1d783649b..d2eb9055ca5 100644
--- a/arch/arm/mach-stm32mp/stm32mp1/stm32mp15x.c
+++ b/arch/arm/mach-stm32mp/stm32mp1/stm32mp15x.c
@@ -223,7 +223,7 @@ void stm32mp_cpu_init(void)
 	if (!IS_ENABLED(CONFIG_SPL) || IS_ENABLED(CONFIG_XPL_BUILD)) {
 		/* Reset Coprocessor state unless it wakes up from Standby power mode */
 		if (!(readl(PWR_MCUCR) & PWR_MCUCR_SBF)) {
-			writel(TAMP_COPRO_STATE_OFF, TAMP_COPRO_STATE);
+			writel(0, TAMP_COPRO_STATE);
 			writel(0, TAMP_COPRO_RSC_TBL_ADDRESS);
 		}
 	}
-- 
2.47.2

